extensions/pkg/controller/csimigration/reconciler.go

```diff
@@ -66,37 +64,41 @@ func (r *reconciler) InjectClient(client client.Client) error {
 }
 
 func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := logf.FromContext(ctx)
+
 	cluster := &extensionsv1alpha1.Cluster{}
 	if err := r.client.Get(ctx, request.NamespacedName, cluster); err != nil {
 		if apierrors.IsNotFound(err) {
+			log.V(1).Info("Object is gone, stop reconciling")
 			return reconcile.Result{}, nil
 		}
-		return reconcile.Result{}, err
+		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
 	}
 
 	shoot, err := extensionscontroller.ShootFromCluster(cluster)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
+	log = log.WithValues("shoot", client.ObjectKeyFromObject(shoot))
+
 	if extensionscontroller.IsShootFailed(shoot) {
-		r.logger.Info("Skipping the reconciliation of csimigration of failed shoot", "csimigration", kutil.ObjectName(shoot))
+		log.Info("Skipping the reconciliation of csimigration of failed shoot")
 		return reconcile.Result{}, nil
 	}
 
 	if cluster.DeletionTimestamp != nil {
 		return reconcile.Result{}, nil
 	}
 
-	r.logger.Info("CSI migration controller got called with cluster", "csimigration", kutil.ObjectName(cluster))
-
-	return r.reconcile(ctx, cluster, shoot)
+	log.Info("CSI migration controller got called with cluster")
```

Comment:
The log now does not include the cluster name and will instead print the shoot object key. Perhaps ot should be changed to:
 ```suggestion
 	log.Info("CSI migration controller got called for shoot")
 ```
 
 Additionally all subsequent logs will print the shoot key instead of the cluster, but I guess that is ok?

Reply:
It will print have both keys (`cluster` because that's the object the controller reconciles, and `shoot` because I added it manually). I think it's still fine like this?

Reply:
Ah yes you're right. I forgot that the CSI migration controller reconciles `cluster` resources. It's fine as it currently is